.gitlab/choco_build/choco_build.yml

```
@@ -42,6 +42,8 @@ windows_choco_online_7_x64:
     - docker run --rm -v "$(Get-Location):c:\mnt" 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/windows_1809_${ARCH}${Env:DATADOG_AGENT_WINBUILDIMAGES_SUFFIX}:${Env:DATADOG_AGENT_WINBUILDIMAGES} c:\mnt\tasks\winbuildscripts\chocopack.bat online
     - If ($lastExitCode -ne "0") { throw "Previous command returned $lastExitCode" }
     - copy build-out\*.nupkg omnibus\pkg
+    - $CopyNupkgToS3 = "$S3_CP_CMD --recursive --exclude '*' --include '*.nupkg' build-out $S3_RELEASE_ARTIFACTS_URI/choco/nupkg"
+    - Invoke-Expression $CopyNupkgToS3
```

Why do you use `Invoke-Expression`? I don't see `S3_CP_CMD` used this way elsewhere.

```suggestion
    - & "$S3_CP_CMD" --recursive --exclude '*' --include '*.nupkg' build-out "$S3_RELEASE_ARTIFACTS_URI/choco/nupkg"
```

---

I used Invoke Expression because powershell doesn't expand the `$S3_CP_CMD` variable correctly. 

I tried using it without the Invoke Expression and got this [error](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/626485354#:~:text=At%20line%3A542,expression%20or%20statement.) ❌ 
This is the [result](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/628076310#:~:text=%24%20%24%20Invoke%2DExpression%20%24CopyNupkgToS3,for%20successful%20job) using the Invoke Expression ✅

---

that error happens when the first token in a command is a string, PowerShell treats it as an expression instead. You must prefix the string with `&` to get powershell to treat it as a command.
for example
```PowerShell
PS C:\Users\branden.clark> "echo" --recursive
ParserError:
Line |
   1 |  "echo" --recursive
     |           ~~~~~~~~~
     | Unexpected token 'recursive' in expression or statement.
PS C:\Users\branden.clark> & "echo" --recursive
--recursive
 ```
I wonder why it isn't necessary elsewhere
https://github.com/DataDog/datadog-agent/blob/7082e7f900e325482ee378df6c9c3c1945c7e14c/.gitlab/deploy_packages/windows.yml#L34-L40

---

I [tested prefixing the command](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/af8a0bd30b2384085b0d8550fb152dd8f97b40b8) with `&` but it didn't work as expected [(results)](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/659222291). After further discussion with @clarkb7, we realized that the issue occurs because `$S3_CP_CMD` contains multiple tokens (`aws s3 cp --no-progress --region us-east-1 --SSE AES256`) and not just a single command like `aws`. For now, using Invoke-Expression seems to be the most suitable option.

---

@julien-lebot thanks for the suggestion. I actually [tested with the changes you mentioned](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/56b063830ec2fa4f3cfbf7666b9ceafcca9910cb) already and unfortunately it didn't work [(results)](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/626485354#:~:text=At%20line%3A542,expression%20or%20statement.)